source/common/router/router.cc

```diff
@@ -750,7 +797,8 @@ Http::FilterDataStatus Filter::decodeData(Buffer::Instance& data, bool end_strea
   // a backoff timer.
   ASSERT(upstream_requests_.size() <= 1);
 
-  bool buffering = (retry_state_ && retry_state_->enabled()) || !active_shadow_policies_.empty() ||
+  bool buffering = (retry_state_ && retry_state_->enabled()) ||
+                   (!active_shadow_policies_.empty() && !streaming_shadows_) ||
```

Comment:
Looking at what this checks gate, it seems like it would clear out `active_shadow_policies_` in the case where we're streaming shadow and (retries or internal redirects) are enabled - is that going to cause a problem?